src/SMAIAXBackend.Application/Services/Implementations/SmartMeterCreateService.cs

```diff
@@ -18,4 +24,28 @@ public async Task<Guid> AddSmartMeterAsync(SmartMeterCreateDto smartMeterCreateD
 
         return smartMeterId.Id;
     }
+
+    public async Task<Guid> AddMetadataAsync(Guid smartMeterId, MetadataCreateDto metadataCreateDto, string? userId)
+    {
+        var validatedUserId = await userValidationService.ValidateUserAsync(userId);
+        var smartMeter =
+            await smartMeterRepository.GetSmartMeterByIdAndUserIdAsync(new SmartMeterId(smartMeterId), validatedUserId);
+
+        if (smartMeter == null)
+        {
+            logger.LogWarning("SmartMeter with id {SmartMeterId} not found for user {UserId}", smartMeterId,
+                validatedUserId.Id);
+            throw new SmartMeterNotFoundException(smartMeterId, validatedUserId.Id);
+        }
+
+        var metadataId = smartMeterRepository.NextMetadataIdentity();
+        var location = new Location(metadataCreateDto.Location.StreetName, metadataCreateDto.Location.City,
```

Comment:
for conversion between dtos and valueobjects/eintities from and to methods can be used